#!/usr/bin/env python

# set up parameters that we care about
PACKAGE = 'jsk_pcl_ros'

try:
    import imp
    imp.find_module(PACKAGE)
    from dynamic_reconfigure.parameter_generator_catkin import *;
except:
    import roslib; roslib.load_manifest(PACKAGE)
    from dynamic_reconfigure.parameter_generator import *;

from math import pi

gen = ParameterGenerator ()
gen.add("min_size", int_t, 0, "the minimum number of the points of each cluster", 100, 0, 10000)
gen.add("distance_threshold", double_t, 0, "distance threshold of organized plane segmentation", 0.05, 0, 1.0)
gen.add("angular_threshold", double_t, 0, "angular threshold of organized plane segmentation", 5.0 / 180.0 * pi, 0, 2 * pi)
gen.add("max_curvature", double_t, 0, "maximum curvature of organized plane segmentation", 0.5, 0, 10.0)
gen.add("connect_plane_angle_threshold", double_t, 0, "plane angle threshold of connecting the planes", 0.1, 0, pi)
gen.add("connect_plane_distance_threshold", double_t, 0, "plane distance threshold of connecting the planes", 0.01, 0, 1.0)
gen.add("connect_distance_threshold", double_t, 0, "distance threshold of connectin the planes", 0.1, 0, 1.0)

exit (gen.generate (PACKAGE, "jsk_pcl_ros", "OrganizedMultiPlaneSegmentation"))

